mlflow/tracking/client.py

```diff
@@ -1016,7 +1036,12 @@ def log_dict(self, run_id, dictionary, artifact_file) -> None:
                     json.dump(dictionary, f, indent=2)
 
     @experimental
-    def log_figure(self, run_id, figure, artifact_file) -> None:
+    def log_figure(
+        self,
+        run_id: str,
+        figure: Union["matplotlib.figure.Figure", "plotly.graph_objects.Figure"],
```

Comment:
Does these string type names "matplotlib.figure.Figure", "plotly.graph_objects.Figure"
works when IDE like pycharm checking type error ?

Reply:
I don't think it works. I think we need to add the following import statements:

```python
try:
    # These packages might not be installed
    import matplotlib
    import ploly
except ImportError:
    ...
```